### DISCUSSION TITLE
Can LittleFS files be accessed from a PC?

### DISCUSSION THREAD

**Discussion by Laserjones** [khoih-prog/LittleFS_Mbed_RP2040#2] [Q&A] [answered]
Hello! I'm very new to this stuff and I was wondering whether files created via LittleFS in the flash memory of a Raspberry Pico can be read by a PC via the Pico's USB port, or (vice versa) files can be uploaded from the PC to LittleFS and accessed from the Pico firmware? If that's possible, what would be required on the PC side to do so?

If it's not possible, is there any other way to make files in the Pico flash memory accessible from both the Pico firmware and the PC?

Best regards,
Joerg

**Comment by khoih-prog** [comment, marked as answer]
Everything is possible, but certainly somebody needs to write some code for it, if not available yet.

Check the examples in my libraries

1.  [**ESP32_FS_EthernetWebServer**](https://github.com/khoih-prog/EthernetWebServer/tree/master/examples/ESP32_FS_EthernetWebServer) and [**FS_EthernetWebServer**](https://github.com/khoih-prog/EthernetWebServer/tree/master/examples/FS_EthernetWebServer) of [**EthernetWebServer** library](https://github.com/khoih-prog/EthernetWebServer), 
2. [**AsyncFSBrowser_WT32_ETH01**](https://github.com/khoih-prog/AsyncWebServer_WT32_ETH01/tree/main/examples/AsyncFSBrowser_WT32_ETH01) of [**AsyncWebServer_WT32_ETH01** library](https://github.com/khoih-prog/AsyncWebServer_WT32_ETH01)
3. [**Async_ESP32_FSWebServer**](https://github.com/khoih-prog/ESPAsync_WiFiManager/tree/master/examples/Async_ESP32_FSWebServer), etc. of [**ESPAsync_WiFiManager** library](https://github.com/khoih-prog/ESPAsync_WiFiManager)
4. [**ESP32_FSWebServer**](https://github.com/khoih-prog/ESP_WiFiManager/tree/master/examples/ESP32_FSWebServer) of [**ESP_WiFiManager** library](https://github.com/khoih-prog/ESP_WiFiManager)

where you can use the PC's  `curl` function to load the software onto ESP32 LittleFS by

```
for file in `\ls -A1`; do curl -F "file=@$PWD/$file" localIPAddress/edit; done
```

So, the easiest way is to use a [**WiFiWebServer**](https://github.com/khoih-prog/WiFiWebServer) or [**EthernetWebServer**](https://github.com/khoih-prog/EthernetWebServer) to generate a full-fledged **WebServer**, then you can **read / write** any **LittleFS** (**SPIFFS, FlashStorage, or SD-based**) file using TCP/IP, Browser, etc..

Just adapt / port the ESP32/ESP8266 code to fulfil your use-case.

I'm sorry I can't help write that code now. Possibly in the future if there are more use-cases and requests. Who knows when.

Good Luck,

**Reply by Laserjones** [reply]
Thank you very much! I had thought of something like plugging the Pico into the PC via USB and accessing it as a mass storage device to see and modify the LittleFS files, but it looks like it's not that easy. ;-)
